### DISCUSSION TITLE
The DARhosimpleCFoam has wrong results when  simulate the actual flight flow conditions .

### DISCUSSION THREAD

**Discussion by NUAA-qinfei** [mdolab/dafoam#690] [General]
I know for  transonic flow (Ma 0.6~1.0) should use DARhosimpleCFoam.  DAFoam have some case for DARhosimpleCFoam , such as DPW4 aircraft, CRM_Wing, NACA0012_Airfoil/transonic and RAE2822_Airfoil ,but the flow conditions of all these cases at sea level conditon, where P0=101325pa,T0=300K,rho0=p0/T0/287.these cases works very well in this condition setup.But when I simulate the flow at  actual flight altitude,such as Altitude = 37,000 ft,where the P0=22700pa,T0=216.774K, the DARhosimpleCFoam has wrong results and the pathline around the DPW4 wing is wrong. I change the P0 ,U0,T0 and nuTilda0 in the runscript.py,and also change the "mu" in constant/thermophysicalProperties to ensure the reynolds number.But the error always happens.
so I want to ask  how to change the flow conditions to  simulate the actual flight conditions,should I change the  system/fvschemes and fvsolution?
this picture shows the DAFoam DPW4 aircraft case ,Ma=0.85,aoa=2.014°，I change the flow condition at  actual flight Altitude = 37,000 ft,where the P0=22700pa,T0=216.774K,rho0=p0/T0/287, nuTilda0=4.07e-5（kinematic viscosity）, mu=1.4875e-5(dynamic viscosity), Re=4.313e7(based on MAC=0.701m).we can see the pathline around the wing slice is not normally,and the results is wrong(CL Cd is wrong.but if I use the p0=101325,t0=300k at the same reynolds number,it works well.All the transonic case in DAFoam have the same problem.
![微信图片_20241003105721_副本](https://github.com/user-attachments/assets/4ec4a8ea-6239-4ce6-a258-c260e485299d)
[logOpt.txt](https://github.com/user-attachments/files/17240814/logOpt.txt)
[runScript_v2.txt](https://github.com/user-attachments/files/17240821/runScript_v2.txt)
[DPW4_AircraftRe4317WD.zip](https://github.com/user-attachments/files/17240884/DPW4_AircraftRe4317WD.zip)

**Comment by friedenhe** [comment]
This is strange. Have you tried to run the case with OpenFOAM's builtin rhoSimpleFOam and see if you see the same problem? Also, other than the streamline, do you see other problems? Like the drag and lift?

**Reply by NUAA-qinfei** [reply]
Thanks for Prof's reply,when I use the  rhoSimpleFOam,it will floating point exception (core dumped) in the Time=1,other than the streamline,the drag and lift is always wrong,CL=-0.2186,Cd=0.08458,the nutilda residual is 0.00219(very high),other residual is normally(below 1e-05),this case is  at 11 km.

**Comment by friedenhe** [comment]
What is the yPlus for the Re~10^7 simulation? We had a similar problem before. Our simulated CL for high-altitude cases was way off (not negative though). We needed to use a yPlus ~30 mesh to ensure realistic simulation results. Another thing you can try is to set `constrainHbyA` to True: https://github.com/mdolab/dafoam/blob/bd386fd0eaa09b46a1d6ea1960545c896147c2c3/dafoam/pyDAFoam.py#L515

**Reply by 123123XU** [reply]
Professor, I'm having the same problem, and unfortunately I've tried the method you said and it doesn't seem to improve.
Could it be that the current solver is unable to solve the aerodynamic simulation calculation at an altitude of 37,000 ft?
Or is there any other good way?
We look forward to your advice.

**Comment by friedenhe** [comment]
Thanks for reporting this. I have converted this to a bug and will let you know once this issue is fixed.

**Reply by NUAA-qinfei** [reply]
Thanks for Professor He's reply,several days ago we find a method to fix this problem ,first :as you say to set constrainHbyA to True .but only do this is not enough ,second :we should also change the range of "pMax" and "pMin" in primalVarBounds .Both of these must be done to make the results normally.only do the first or only do the second is not enough. (the mesh we use yPlus ~30) I think this phenomenon is related to SIMPLE method to slove compressible problems ,even the OpenFoam original solver rhoSimpleFoam have the similar problem ,when the environmental density is too small ,the pressure correction equation may not useful,so the openFoam have the rhoCentralFoam and sonicFoam to use density-based solver. The two pictures show our change for the second and our preliminary test results,then we will continue to test. Thanks for Professor He, your suggestion for the first point is crucially important.
<img width="886" height="1032" alt="微信图片_20250719223141" src="https://github.com/user-attachments/assets/066e709b-6989-42eb-9cfa-a2645f111e9c" />
![微信图片_20250719225023](https://github.com/user-attachments/assets/f9080c7e-10ec-498f-bdf4-44ad9edcf65e)

**Comment by friedenhe** [comment]
It is great to hear that changing constrainHbyA and pMax, pMin bounds improved the simulation! I am not too surprised to see the difference among OpenFOAM, StarCCM, and Fluent. I agree that it is not an optimal option to use pressure-based solvers, such as DARhoSimpleCFoam, to simulate high-altitude, transonic flow. Keep us posted!
